### DISCUSSION TITLE
How to come up with proof of code

### DISCUSSION THREAD

**Discussion by guyzilberblum** [Cyfrin/security-and-auditing-full-course-s23#170] [General]
I Think i have a hard time to come up with proof of code im in the puppy raffle section and the mission now is to come up with poc to overflow and i would never thought to solve the proof of code like patrick did does any one also have hard time to come up with proof of code because im teribble at this and i dont know how cna i improve it thanks in advance

**Comment by EngrPips** [comment]
Hello @guyzilberblum, writing tests for complex systems can really be hard, but it will get easier with time. I would say just keep trying on your own and when you fail look at how Patrick did it and learn what you got wrong and what you didn't properly think of. Doin this process repeatedly would help you improve immensly.

**Comment by guyzilberblum** [comment]
Thanks for the tip!

**Reply by EngrPips** [reply]
Anytime. One thing I would mention is to make sure to understand the system perfectly before attempting to write a test for it.

**Comment by guyzilberblum** [comment]
Thanks, I will!

**Reply by EngrPips** [reply]
You are welcome, Friend.
